### PR TITLE
Add Home Type filter

### PR DESCRIPTION
**Pull request by woeltjen**
## Description
Adds the "Home Type" filter from updated Beenest design

## How to Test
1. Search for some listings (e.g. in San Francisco, where we have many kinds)
2. Click "Home Type"
3. Choose a different home type
4. Expect listings to filter down to that home type
5. Regression test Destination filters

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
This exposes some UX issues that we may want to solve now, or carry:

1. Approach to dropdown width from #304 was broken (it depended on the dropdown button being in a full-width column, but that space is needed for more buttons). Solution here (explicit widths for dropdowns in rems) is not mobile friendly.
2. When a Home Type filter results in listings, the "No listings were found..." page displays, losing the context of the original search.

### CODE REVIEW

File: src/pages/search/SearchForm.tsx [hunk 1]
```diff
@@ -13,36 +14,24 @@ interface Props {
...
-      <Col>
-        <SearchFilter label="Add Destination">
+      <ButtonGroup>
+        <SearchFilter label="Home Type" width="9rem">
+          <HomeType
+            homeType={filter.homeType}
+            onChange={homeType => onFilterChange({ ...filter, homeType })}
+          />
+        </SearchFilter>
+        <SearchFilter label="Destination" width="24rem">
```
Comment (woeltjen) [inline src/pages/search/SearchForm.tsx]: "Add Destination" looked inconsistent next to "Home Type" (verb versus noun); with multiple buttons, I think the context is clear enough that just "Destination" is fine here, but want to call that out for review

Comment (kcvan) [reply src/pages/search/SearchForm.tsx]: i think now that we have the home type filter next to it 'Destination' seems good now!

**Comment by woeltjen** [comment]
> works! can we add `noGutters` to the row that the buttons are in?

Ah, thanks, I knew there was something off about the alignment there. Added!
